components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -0,0 +1,202 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCBannerView.h"
+
+#import "MDCBannerParams.h"
+#import "MDCBannerViewLayout.h"
+#import "MaterialColorScheme.h"
+#import "MaterialTypography.h"
+
+@interface MDCBannerView () <MDCBannerViewLayoutDataSource>
+
+@property(nonatomic, readwrite, weak) UILabel *textLabel;
+@property(nonatomic, readwrite, weak) UIImageView *iconImageView;
+@property(nonatomic, readwrite, strong) NSMutableArray<__kindof UIButton *> *buttons;
+
+@property(nonatomic, readwrite, strong) MDCBannerViewLayout *layout;
+
+@end
+
+@implementation MDCBannerView
+
+- (instancetype)initWithFrame:(CGRect)frame {
```

Comment:
Consider using the factory Banner method as the correct way to create a banner instance, and marking default initializers as NS_UNAVAILABLE. It might be easier than handling all the edge cases in which some of the data points is unavailable when the banner is presented.

Reply:
Actionable Things: add example that use bannerView in storyboard, which requires the default initializer.

Reply:
Ticketed in b/118629684.